core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -268,6 +271,11 @@ public Builder withEncryptionKeyMetadata(EncryptionKeyMetadata newKeyMetadata) {
       return withEncryptionKeyMetadata(newKeyMetadata.buffer());
     }
 
+    public Builder withSortOrderId(int newSortOrderId) {
```

Comment:
What about `withSortOrder`? Is there a case where we would only have the ID and not the `SortOrder`? Just trying to think of what will be convenient. We could always expose both methods.

Reply:
I think if we do that, [`copy()`](https://github.com/apache/iceberg/pull/1975/files#diff-ac5b4d1edab36f13fd929683cad0f56951ac019353354ac886e84d099d5dd219R179) of this Builder might have problem since `DataFile` only has `sortOrderId()` but no `sortOrder()` so we couldn't copy directly; we probably also couldn't add `sortOrder()` to `ContentFile`/`DataFile` since the file created from Avro reflection will not have the struct without looking up table metadata?